tokio-console/src/main.rs

```diff
@@ -214,3 +223,24 @@ async fn watch_details_stream(
         }
     }
 }
+
+fn gen_completion(install: bool, shell: Shell) -> color_eyre::Result<()> {
+    let mut app = config::Config::command();
+    let mut buf: Box<dyn std::io::Write> = if install {
+        let mut home_dir = dirs::home_dir()
+            .ok_or_else(|| color_eyre::eyre::eyre!("fail to find home directory"))?;
+        match shell {
+            Shell::Zsh => {
+                home_dir.push(".zsh_functions/_tokio_console");
+                let f = std::fs::File::create(&home_dir)
+                    .wrap_err_with(|| format!("fail to open {}", home_dir.display()))?;
+                Box::new(std::io::BufWriter::new(f))
+            }
+            _ => color_eyre::eyre::bail!("Not support to install completion script on {}", shell),
+        }
+    } else {
+        Box::new(std::io::stdout())
+    };
+    generate(shell, &mut app, "tokio-console", &mut buf);
+    Ok(())
+}
```

Comment:
@hawkw 
How about this approach?
If the user runs as `tokio-console gen-completion --install <SHELL>` then put the script file.

But I must fix it, `.zsh_functions` directory is not correct. I have a problem.
I have tried to get a directory with `$FPATH` environment variables however I failed.

problem: 
`std::env::var("FPATH")` return an error. I checked on my terminal like this:

```
❯ echo $FPATH
/home/nrskt/.zsh_functions:....some paths

❯ env | rg FPATH
(return nothing)
```
So `FPATH` not working in my environment. I can't determine if it's a problem with my environment or zsh features.

---

In my opinion, first I fix it to not support all shells like this:

```
match shell {
    _ => color_eyre::eyre::bail!("Not support to install completion script on {}", shell),
}
```

Reply:
hmm...it seems like automatically installing completions may be a bit more complicated than i'd hoped. i think it's reasonable to just always fail if `--install` is passed, for now, so we can go ahead and merge this branch, and implement support for automatic installations later.